src/gui/qgsfeatureselectiondlg.cpp

```diff
@@ -128,3 +131,14 @@ void QgsFeatureSelectionDlg::mActionPanMapToSelectedRows_triggered()
 {
   mContext.mapCanvas()->panToSelected( mVectorLayer );
 }
+
+void QgsFeatureSelectionDlg::setFilterExpression( const QString &filter, QgsAttributeForm::FilterType type )
+{
+  mFeatureFilterWidget->setFilterExpression( filter, type, true );
+}
+
+void QgsFeatureSelectionDlg::viewModeChanged( QgsAttributeEditorContext::Mode mode )
+{
+  if ( mode != QgsAttributeEditorContext::SearchMode )
+    mActionSearchForm->setChecked( false );
```

Comment:
Does this not need to be changed in the other direction too?

```suggestion
    mActionSearchForm->setChecked( mode == QgsAttributeEditorContext::SearchMode );
```

Reply:
I don't think so, because mActionSearchForm is already checked. The only way to switch to SearchMode is to toggle mActionSearchForm

Reply:
In case someone adds a keyboard shortcut or other fancy way to switch this behavior, would it hurt to make sure the state is synchronized?

Reply:
I was afraid that it triggers another useless signal even if the state didn't change, but it's not the case according to doc and my tests. I made the modifications.

Reply:
For reference, when I write properties, I normally do the following (and I think I started when I read somewhere Qt handles it the same way to avoid binding loops):

```
void setProperty( const Property &property )
{
  if ( property == mProperty )
    return;

  mProperty = property;
  emit propertyChanged();
}
```

Reply:
Good to know. But I think it not the case for QLineEdit::setText for instance which fire a textChanged on every setText even if the text hasn't changed